lib/ansible/module_utils/vmware.py

```diff
@@ -234,11 +234,11 @@ def compile_folder_path_for_object(vobj):
     thisobj = vobj
     while hasattr(thisobj, 'parent'):
         thisobj = thisobj.parent
+        if thisobj.name == 'Datacenters':
```

Comment:
@dagwieers @pdellaert 

This while loop is trying to build path for a object. We need to break when we encounter 'Datacenters' which is parent of every object. In Python2*, it works perfectly, but in Python3, it fails. 
So I added `break` when we encounter Datacenters, since we don't have Datacenters in paths list then, there is no point in keeping `if paths[0] == 'Datacenters':` statement.
Without fix [this script](https://github.com/Akasurde/pyvmomi_examples/blob/master/compile_path_example.py) fails
With fix: 
```
# python compile_path_example.py
/
/vm
/datastore
/network
/
# python3 compile_path_example.py
/
/vm
/datastore
/network
/
# govc find /
/
/DC0
/DC0/vm
/DC0/vm/DC0_H0_VM0
/DC0/vm/DC0_H0_VM1
/DC0/vm/DC0_C0_RP0_VM0
/DC0/vm/DC0_C0_RP0_VM1
/DC0/host
/DC0/host/DC0_H0
/DC0/host/DC0_H0/DC0_H0
/DC0/host/DC0_H0/Resources
/DC0/host/DC0_C0
/DC0/host/DC0_C0/DC0_C0_H0
/DC0/host/DC0_C0/DC0_C0_H1
/DC0/host/DC0_C0/DC0_C0_H2
/DC0/host/DC0_C0/Resources
/DC0/datastore
/DC0/datastore/LocalDS_0
/DC0/network
/DC0/network/VM Network
/DC0/network/DVS0
/DC0/network/DC0_DVPG0
```